### PR TITLE
[AKS] bring in support for cluster autoscaler profile in core CLI

### PR DESCRIPTION
**Pull request by marwanad**
**Description**  

This brings in support for setting cluster autoscaler profile. The support already exists in aks-preview extension so that's just bringing support to core CLI.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 1]
```diff
@@ -118,6 +120,35 @@ def validate_k8s_version(namespace):
...
+        for key in namespace.cluster_autoscaler_profile.keys():
+            _validate_cluster_autoscaler_key(key)
+
+
+def _validate_cluster_autoscaler_key(key):
+    if not key:
+        raise CLIError('Empty key specified for cluster-autoscaler-profile')
+    valid_keys = list(k.replace("_", "-") for k, v in ManagedClusterPropertiesAutoScalerProfile._attribute_map.items())  # pylint: disable=protected-access
+    if key not in valid_keys:
+        raise CLIError('Invalid key specified for cluster-autoscaler-profile: %s' % key)
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py:140]: Can we also list all the valid keys in the help message ?

Comment (marwanad) [reply src/azure-cli/azure/cli/command_modules/acs/_validators.py]: good call, updated.

File: linter_exclusions.yml [hunk 2]
```diff
@@ -217,6 +217,9 @@ aks create:
     enable_cluster_autoscaler:
       rule_exclusions:
       - option_length_too_long
+    cluster_autoscaler_profile:
+      rule_exclusions:
+      - option_length_too_long
```
Comment (MyronFanQiu) [inline linter_exclusions.yml:222]: Please add one more short abbreviation for your new argument instead of exclude this rule

Comment (marwanad) [reply linter_exclusions.yml:222]: I don't think there's a reasonable short option for this. This exists as this in the CLI extensions and we'd like to keep it this way.

Comment (MyronFanQiu) [reply linter_exclusions.yml:222]: a long argument is not user friendly. You can keep this one same with extension. One argument can have several options.

Comment (marwanad) [reply linter_exclusions.yml:222]: sounds reasonable to me, will make the change

**Comment by arrownj** [comment]
Hi @marwanad , could you please resolve the conflict then I will merge it ?

**Comment by marwanad** [comment]
@arrownj Rebased! Test failure seems unrelated.

**Pull request merged**